### ISSUE TITLE
Comparison to `nix-serve-ng`

### ISSUE THREAD

**Issue by srid** [nix-community/harmonia#335]
> Harmonia is a binary cache for nix that serves your /nix/store as a binary cache over http. It's written in Rust **for speed**.

[nix-serve-ng's benchmarks](https://github.com/aristanetworks/nix-serve-ng?tab=readme-ov-file#benchmarks) indicate that `harmonia` is somewhat slower than `nix-serve-ng`, notably for fetching 10+ MB NARs (harmonia is slower than nix-serve itself here).

<img width="674" alt="image" src="https://github.com/nix-community/harmonia/assets/3998/85270e4e-cddc-4f56-9723-aa5167090715">

Are these benchmarks accurate for the code in `master` branch?

In any case, mentioning how this project compares (pros? cons?) to `nix-serve-ng` in the README would be useful.

**Comment by Mic92** [comment]
No. They are not up-to-date. Many optimization have landed since than in harmonia.

**Comment by Mic92** [comment]
Here you can get an up-to-date benchmark: https://github.com/Mic92/cache-shootout/
